core/src/main/java/hudson/ClassicPluginStrategy.java

```diff
@@ -629,8 +629,17 @@ private void updateTransientDependencies() {
             if (PluginManager.FAST_LOOKUP) {
                 for (PluginWrapper pw : getTransitiveDependencies()) {
                     try {
-                        Class<?> c = ClassLoaderReflectionToolkit._findLoadedClass(pw.classLoader, name);
+                        Class<?> c;
+                        if (pw.classLoader instanceof AntWithFindResourceClassLoader) {
+                            c = ((AntWithFindResourceClassLoader) pw.classLoader).findLoadedClass2(name);
+                        } else {
+                            c = ClassLoaderReflectionToolkit._findLoadedClass(pw.classLoader, name);
```

Comment:
wondering if this `ClassLoaderReflectionToolkit` could be removed, assuming everything goes through `AntWithFindResourceClassLoader` then this line never gets called.

(from my minimal plugin set in my test Jenkins this assumption is always true)

any thoughts? this comment applies throughout the PR

Reply:
Not sure. May apply in `JenkinsRule` under exotic conditions?

Reply:
it does seem to apply in Jenkinsfile :(, see my happiness in my commit message here:
https://github.com/jenkinsci/jenkins/pull/5110/commits/555e516ea72e3319ea06caca0facb7740c1caf48

Reply:
:man_shrugging: if you can verify that only `AntWithFindResourceClassLoader` is ever used at this point, including in various test harnesses, we could probably simplify a bunch of code, and put in some `assert`ions.

Reply:
Well I can verify at runtime it's fine, but in tests (at least AssetManagerTest it's not fine, it runs with a classloader built into the JDK, AppClassLoader I think.

Reply:
So sounds like we still need `ClassLoaderReflectionToolkit` at least for now.